### ISSUE TITLE
BUG: Python's PprzMessage doesn't cast values to the specified types

### ISSUE THREAD

**Issue by MJafarMashhadi** [paparazzi/pprzlink#124]
`msg[value_name]` is always of type `str` and the `fieldtypes` property is ignored there (at least for numbers). It's only used when converting to/from binary.

**Comment by gautierhattenberger** [comment]
Yes, it is true that data in message fields have the same type than the element stored inside. So when they are filled from IVY strings, I guess they all end up as `str` as well.
It's not really a bug, nor a feature... Mostly laziness I guess :(
If you have a clean solution for that, it could be integrated, as there is no reason to store data with the wrong type. And that would avoid explicit cast when extracting them on user side (like here: https://github.com/paparazzi/paparazzi/blob/master/sw/ground_segment/python/multi/fc_rotor/fc_rotor.py#L116)

**Comment by MJafarMashhadi** [comment]
Explicit casting was exactly what I had to do [as well](https://github.com/MJafarMashhadi/pprz_tester/blob/4f6b9f1615d17e47f935a3462454c4886e8691d8/pprz_tester/aircraft.py#L88)!

I think doing the type casting right in [`ivy_string_to_payload`](https://github.com/paparazzi/pprzlink/blob/9cd22b04722560f9299d70091d195c0519b3a98c/lib/v1.0/python/pprzlink/message.py#L181) and `binary_to_payload` would do the job. Also it'd be nicer to have a dictionary mapping names to a 3-tuple of value, dtype, and coef instead of keeping 4 separate lists in-sync for name, value, dtype, coef.
